kong/dao/schemas/apis.lua

```diff
@@ -5,12 +5,14 @@ local fmt = string.format
 local sub = string.sub
 local match = string.match
 
+local PROTOCOLS = { "HTTP", "HTTPS" }
+
 local function validate_upstream_url_protocol(value)
   local parsed_url = url.parse(value)
   if parsed_url.scheme and parsed_url.host then
-    parsed_url.scheme = parsed_url.scheme:lower()
-    if not (parsed_url.scheme == "http" or parsed_url.scheme == "https") then
-      return false, "Supported protocols are HTTP and HTTPS"
+    parsed_url.scheme = parsed_url.scheme:upper()
+    if not utils.table_contains(PROTOCOLS, parsed_url.scheme) then
+      return false, "Supported protocols are: "..table.concat(PROTOCOLS, ", ")
```

Comment:
It's iterating over the table, and hence horribly slow. Just create a reverse indexed table for those lookups, so much faster...